### PR TITLE
Maker: Remove sensor start() methods from toolbox

### PR DESCRIPTION
**Pull request by islemaster**
Removes `soundSensor.start()` and `lightSensor.start()` from the default App Lab toolbox and from droplet autocomplete.  The blocks are still supported, to make sure we don't break any levels that are using them, but they're not needed anymore (sensors are on by default).

### CODE REVIEW

**Comment by aoby** [comment]
LGTM, but I am unfamiliar with these areas of the codebase. You may want another set of eyes.

File: apps/src/lib/kits/maker/dropletConfig.js [hunk 1]
```diff
@@ -114,13 +114,13 @@ export const blocks = [
...
 
-  {func: 'soundSensor.start', category: CIRCUIT_CATEGORY},
+  {func: 'soundSensor.start', category: CIRCUIT_CATEGORY, noAutocomplete: true},
   {func: 'soundSensor.value', category: CIRCUIT_CATEGORY, type: 'readonlyproperty'},
   {func: 'soundSensor.getAveragedValue', category: CIRCUIT_CATEGORY, params: ['500'], paletteParams: ['ms'], type: 'value'},
   {func: 'soundSensor.setScale', category: CIRCUIT_CATEGORY, params: ['0', '100'], paletteParams: ['low', 'high']},
   {func: 'soundSensor.threshold', category: CIRCUIT_CATEGORY, type: 'property' },
 
-  {func: 'lightSensor.start', category: CIRCUIT_CATEGORY},
+  {func: 'lightSensor.start', category: CIRCUIT_CATEGORY, noAutocomplete: true},
```
Comment (islemaster) [inline apps/src/lib/kits/maker/dropletConfig.js:123]: That just means it's knowledge-share time!  Allow me to give you a tour of these changes.

Here I've modified entries in the `maker/dropletConfig.js` `blocks` export.  We have several `dropletConfig.js` files for different apps and different APIs - for example, [here's the App Lab droplet config](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/dropletConfig.js).  These are our own internal format, but many of these options get passed down to the [droplet editor](https://github.com/droplet-editor/droplet) used by App Lab, Game Lab, etc. These configuration files are generally responsible for:

- Providing a set of "blocks" the droplet editor will recognize, with various predefined block properties.
- Defining categories for the toolbox.
- Listing other 'predefined' values that affect linting within the editor.
- A few other configuration options (`showExamplesLink` or `showParamDropdowns`).

Note that this does _not_ determine which blocks appear in the level's toolbox, although this is a prerequisite for it - that happens later.  Nor does this config define the _behavior_ of these blocks, which is an interpreter concern more than an editor concern.

Here we're modifying the configuration of a couple of blocks.  Block configuration options [are documented in dropletUtils.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/dropletUtils.js#L3-L26).  The `noAutocomplete` option that I'm setting here reads "Do not include this function in our ace completer."  [Ace](https://ace.c9.io/) is the text editor that powers Droplet's text mode.  There's no autocomplete in block mode, for obvious reasons.

The droplet config for Maker Toolkit, since it's not associated with a particular app type, [gets "mixed in" to the App Lab droplet configuration](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/applab.js#L494-L494) during App Lab initialization.  We do this even when Maker is disabled, so that the droplet representation of Maker code is always the same.

Comment (aoby) [reply apps/src/lib/kits/maker/dropletConfig.js:123]: Thanks for the detailed description! 
Knowledge has been shared :)

So does this change not remove these blocks from the toolbox, as the subject would suggest?
And are the below changes to shared_constants just removing constants that are no longer needed?

File: lib/cdo/shared_constants.rb [hunk 2]
```diff
@@ -217,11 +217,9 @@ module SharedConstants
       "accelerometer.getAcceleration": null,
       "isPressed": null,
       "holdtime": null,
-      "soundSensor.start": null,
       "soundSensor.value": null,
       "soundSensor.getAveragedValue": null,
       "soundSensor.setScale": null,
-      "lightSensor.start": null,
```
Comment (islemaster) [inline lib/cdo/shared_constants.rb:224]: Deleting these two lines is what removes the `soundSensor.start()` and `lightSensor.start()` methods from the App Lab default toolbox.  You'll notice the keys match the `func` keys given in the droplet config.

This `APPLAB_BLOCKS` list is a constant we share across our Ruby and JavaScript code - mostly for levelbuilder purposes.

In Ruby, it gets used [in the `Applab` model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/applab.rb#L128-L130) to [populate the default toolbox when creating a new App Lab level on levelbuilder](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/applab.rb#L77).

To use this constant in JavaScript we [run generateSharedConstants.rb before every build](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/Gruntfile.js#L701) to [generate an untracked file](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/script/generateSharedConstants.rb#L37) called sharedApplabBlocks.js.  It gets loaded in applab/levels.js and [used as the blocks for the default App Lab level](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/levels.js#L35-L41), which means any level without a palette (`codeFunctions`) will get this one - that includes the [New App Lab Project.level](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/scripts/levels/New%20App%20Lab%20Project.level) which doesn't have a `codeFunctions` entry on purpose.

In summary: This list populates the toolbox for App Lab projects and for new App Lab levels on levelbuilder by referencing keys in the droplet config.

**Comment by islemaster** [comment]
And if that all sounds needlessly convoluted... you might be right! :grinning:

**Comment by islemaster** [comment]
> I have another question. How are the toolbox blocks organized into sections?

Let's use App Lab's droplet config as an example.  Toolbox categories are [defined and exported in a `categories` object](https://github.com/code-dot-org/code-dot-org/blob/v2017-04-18.0/apps/src/applab/dropletConfig.js#L191-L222).  Their definition gives them a category color and default block color, like this:
```jsx
  'UI controls': {
    id: 'uicontrols',
    color: 'yellow',
    rgb: color.droplet_yellow,
    blocks: []
  },
```

Then, a block's definition will include a category key.  For example, [here's the block definition for `textInput()`](https://github.com/code-dot-org/code-dot-org/blob/v2017-04-18.0/apps/src/applab/dropletConfig.js#L72):
```jsx
{func: 'textInput', parent: api, category: 'UI controls', paletteParams: ['id','text'], params: ['"id"', '"text"'] },
```
